Ignore unused textures

Shii2 opened the pull request:
Ignore textures not used by shader and not included into asset bundles

Thryrallo commented:
Are they not included in the build though ?
Cause i asked someone knowledgeable & to their knowledge it is included.
Do you have any more info / documentation backing this up?

Shii2 commented:
I tested this in editor. Made a simple cube avatar, added material and set it to UTC shader then assigned main texture and shade map texture then made a build and examined build log. Shade map indeed exists in asset list.
Then I switched shader to Poiyomi (it does not support shade map material property) and made another build then examined log again - shade map texture now missing from the list.
Shade map texture: **Assets/Avatars/KissMa/_CommonFolder/Textures/hair1_mask4.png**

Log output:
```
UTC Shader with shade map texture hair1_mask4.png
-------------------------------------------------------------------------------
Bundle Name: prefab-id-v1_avtr_3d2543ce-8dd9-48bf-a227-561e6e25b8d2_0592254213.prefab.unity3d
Compressed Size:243.1 kb
Uncompressed usage by category (Percentages based on user generated assets only):
Textures               1.5 mb	 66.9% 
Meshes                 0.0 kb	 0.0% 
Animations             0.0 kb	 0.0% 
Sounds                 0.0 kb	 0.0% 
Shaders                635.6 kb	 27.7% 
Other Assets           43.2 kb	 1.9% 
Levels                 0.0 kb	 0.0% 
Scripts                0.2 kb	 0.0% 
Included DLLs          0.0 kb	 0.0% 
File headers           80.6 kb	 3.5% 
Total User Assets      2.2 mb	 100.0% 
Complete build size    2.2 mb
Used Assets and files from the Resources folder, sorted by uncompressed size:
 1.0 mb	 44.6% Assets/Avatars/KissMa/UltimateSporty/Textures/faceparts.png
 635.6 kb	 27.7% Assets/Toon/Shader/Toon_DoubleShadeWithFeather.shader
 512.2 kb	 22.3% Assets/Avatars/KissMa/_CommonFolder/Textures/hair1_mask4.png
 40.2 kb	 1.8% Assets/TestMaterial.mat
 2.4 kb	 0.1% Assets/prefab-id-v1_avtr_3d2543ce-8dd9-48bf-a227-561e6e25b8d2_0592254213.prefab
 0.5 kb	 0.0% AssetBundle Object
 0.1 kb	 0.0% Assets/VRCSDK/Plugins/VRCSDK3A.dll
 0.1 kb	 0.0% Assets/VRCSDK/Plugins/VRCCore-Editor.dll
-------------------------------------------------------------------------------

Poiyomi shader, shade map texture hair1_mask4.png referenced in material but missing from the build
-------------------------------------------------------------------------------
Bundle Name: prefab-id-v1_avtr_3d2543ce-8dd9-48bf-a227-561e6e25b8d2_4587256752.prefab.unity3d
Compressed Size:265.0 kb
Uncompressed usage by category (Percentages based on user generated assets only):
Textures               1.1 mb	 70.2% 
Meshes                 0.0 kb	 0.0% 
Animations             0.0 kb	 0.0% 
Sounds                 0.0 kb	 0.0% 
Shaders                359.3 kb	 21.9% 
Other Assets           47.5 kb	 2.9% 
Levels                 0.0 kb	 0.0% 
Scripts                0.2 kb	 0.0% 
Included DLLs          0.0 kb	 0.0% 
File headers           82.4 kb	 5.0% 
Total User Assets      1.6 mb	 100.0% 
Complete build size    1.6 mb
Used Assets and files from the Resources folder, sorted by uncompressed size:
 1.0 mb	 62.3% Assets/Avatars/KissMa/UltimateSporty/Textures/faceparts.png
 359.3 kb	 21.9% Assets/OptimizedShaders/TestMaterial/S_Poiyomi_Pro.shader
 128.4 kb	 7.8% Assets/_PoiyomiShaders/Textures/Defaults/T_Default_CM.exr
 44.6 kb	 2.7% Assets/TestMaterial.mat
 2.4 kb	 0.1% Assets/prefab-id-v1_avtr_3d2543ce-8dd9-48bf-a227-561e6e25b8d2_4587256752.prefab
 0.9 kb	 0.1% Assets/_PoiyomiShaders/Textures/Defaults/T_Default_SR.png
 0.5 kb	 0.0% AssetBundle Object
 0.1 kb	 0.0% Assets/VRCSDK/Plugins/VRCSDK3A.dll
 0.1 kb	 0.0% Assets/VRCSDK/Plugins/VRCCore-Editor.dll
-------------------------------------------------------------------------------
```

Thryrallo commented:
This is probably correct. It seems like non-referenced textures are most likely not added to the build. 
However, I am not 100% sure on this, as some other sources point to this not being the case. In that case it still shouldn't affect the VRAM, but is something I would like to keep my eyes open for.
